galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerContainerResourceMonitor.java

```diff
@@ -0,0 +1,163 @@
+package dev.galasa.docker.internal;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.gson.Gson;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResourceManagement;
+import dev.galasa.http.HttpClientException;
+import dev.galasa.http.HttpClientResponse;
+import dev.galasa.http.IHttpClient;
+import dev.galasa.http.StandAloneHttpClient;
+
+/**
+ * Resource monitor for cleaning up orphaned containers.
+ * 
+ * @author James Davies
+ */
+public class DockerContainerResourceMonitor implements Runnable {
+    private final IFramework framework;
+    private final IResourceManagement resourceManagement;
+    private final IConfigurationPropertyStoreService cps;
+    private final IDynamicStatusStoreService dss;
+
+    private final Log logger = LogFactory.getLog(DockerContainerResourceMonitor.class);
+
+    private Gson gson = new Gson();
+
+    private Map<String,IHttpClient> dockerEngines = new HashMap<>();
+
+    public DockerContainerResourceMonitor(IFramework framework, IResourceManagement resourceManagement, IConfigurationPropertyStoreService cps, IDynamicStatusStoreService dss) {
+        this.framework = framework;
+        this.resourceManagement = resourceManagement;
+        this.cps = cps;
+        this.dss = dss;
+
+        dockerEngines = getDockerEngines();
```

Comment:
you can't do this in the constructor,   you must obtain the docker engines everytime.
If someone adds a new dockerengine,  whilst it is running,  it won't be picked up.   We should not require people to bounce the resmon everytime they change the cps

Reply:
100%, my bad on that one